### PR TITLE
Console.WriteLine for NetCore Projects

### PR DESCRIPTION
**Pull request by jayaranigarg**

### CODE REVIEW

File: scripts/Build.ps1 [hunk 1]
```diff
@@ -170,6 +172,21 @@ function Perform-Restore {
...
+  
+  $msbuild = Join-Path $msbuildPath "MSBuild.exe"
+  
+  Write-Verbose "Starting restore for NetCore Projects"
+  foreach($project in $TFB_NetCoreProjects)
+  {
+	$projectPath = Locate-Item -relativePath $project
+
+	Write-Verbose "$msbuild /t:restore -verbosity:quiet $projectPath"
+	& $msbuild /t:restore -verbosity:quiet $projectPath
```
Comment (AbhitejJohn) [inline scripts/Build.ps1]: >quiet [](start = 34, length = 5)

can we make this minimal so we know whats going on instead of it looking like its stuck. Might want to change it on top(nuget restore) too.

File: scripts/build/TestFx.Settings.targets [hunk 2]
```diff
@@ -3,6 +3,7 @@
   <PropertyGroup>
     <TestFxRoot Condition="$(TestFxRoot) == ''">..\..\</TestFxRoot>
     <RepoRoot>..\..\</RepoRoot>
+	<CommonRoot>$(TestFxRoot)src\Adapter\Common\</CommonRoot>
```
Comment (AbhitejJohn) [inline scripts/build/TestFx.Settings.targets]: >CommonRoot [](start = 2, length = 10)

Do we need this?

File: src/Adapter/PlatformServices.Shared/netstandard1.0/ns10Constants.cs [hunk 3]
```diff
@@ -3,10 +3,15 @@
 
 namespace Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices
 {
+#pragma warning disable SA1649 // SA1649FileNameMustMatchTypeName
+
     internal class Constants
```
Comment (AbhitejJohn) [inline src/Adapter/PlatformServices.Shared/netstandard1.0/ns10Constants.cs]: >Constants [](start = 19, length = 9)

why not just call this file constants.

File: src/Adapter/PlatformServices.Shared/netstandard1.0/Services/ns10TraceLogger.cs [hunk 4]
```diff
@@ -3,6 +3,7 @@
 
 namespace Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices
 {
+    using System;
```
Comment (AbhitejJohn) [inline src/Adapter/PlatformServices.Shared/netstandard1.0/Services/ns10TraceLogger.cs]: >using System; [](start = 4, length = 13)

is this needed?

File: test/UnitTests/PlatformServices.Shared.Unit.Tests/netstandard1.0/ns10FileOperationsTests.cs [hunk 5]
```diff
@@ -45,7 +51,7 @@ public void LoadAssemblyShouldThrowExceptionIfFileIsNotFound()
         [TestMethod]
         public void LoadAssemblyShouldLoadAssemblyInCurrentContext()
         {
-            var filePath = Assembly.GetExecutingAssembly().Location;
+            var filePath = typeof(FileOperationsTests).GetTypeInfo().Assembly.Location; // Assembly.GetExecutingAssembly().Location;
```
Comment (AbhitejJohn) [inline test/UnitTests/PlatformServices.Shared.Unit.Tests/netstandard1.0/ns10FileOperationsTests.cs]: >// Assembly.GetExecutingAssembly().Location; [](start = 88, length = 44)

nit: remove.

File: test/UnitTests/PlatformServices.Shared.Unit.Tests/netstandard1.0/ns10FileOperationsTests.cs [hunk 6]
```diff
@@ -99,7 +105,7 @@ public void CreateNavigationSessionShouldReturnDiaSession()
                     typeof(MockDiaSession).AssemblyQualifiedName,
                     typeof(MockDiaNavigationData).AssemblyQualifiedName);
 
-                var diaSession = this.fileOperations.CreateNavigationSession(Assembly.GetExecutingAssembly().Location);
+                var diaSession = this.fileOperations.CreateNavigationSession(typeof(FileOperationsTests).GetTypeInfo().Assembly.Location); // Assembly.GetExecutingAssembly().Location);
```
Comment (AbhitejJohn) [inline test/UnitTests/PlatformServices.Shared.Unit.Tests/netstandard1.0/ns10FileOperationsTests.cs]: >// Assembly.GetExecutingAssembly().Location); [](start = 139, length = 45)

nit:remove. Here and everywhere else.

File: test/UnitTests/PlatformServices.Universal.Unit.Tests/PlatformServices.Universal.Unit.Tests.csproj [hunk 7]
```diff
@@ -35,20 +35,47 @@
     <WarningLevel>4</WarningLevel>
   </PropertyGroup>
   <ItemGroup>
+    <Reference Include="Castle.Core, Version=4.0.0.0, Culture=neutral, PublicKeyToken=407dd0808d44fbdc, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Castle.Core.4.0.0\lib\net45\Castle.Core.dll</HintPath>
+      <Private>True</Private>
```
Comment (AbhitejJohn) [inline test/UnitTests/PlatformServices.Universal.Unit.Tests/PlatformServices.Universal.Unit.Tests.csproj:40]: please remove.

File: test/UnitTests/PlatformServices.Universal.Unit.Tests/PlatformServices.Universal.Unit.Tests.csproj [hunk 8]
```diff
@@ -35,20 +35,47 @@
...
+    </Reference>
     <Reference Include="Microsoft.VisualStudio.QualityTools.UnitTestFramework, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
       <Aliases>FrameworkV1</Aliases>
     </Reference>
+    <Reference Include="Moq, Version=4.7.8.0, Culture=neutral, PublicKeyToken=69f491c39445e920, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Moq.4.7.8\lib\net45\Moq.dll</HintPath>
+      <Private>True</Private>
```
Comment (AbhitejJohn) [inline test/UnitTests/PlatformServices.Universal.Unit.Tests/PlatformServices.Universal.Unit.Tests.csproj:47]: please remove.

File: test/UnitTests/PlatformServices.Universal.Unit.Tests/Services/UniversalTestSourceHostTests.cs [hunk 9]
```diff
@@ -1,59 +0,0 @@
...
-    extern alias FrameworkV1;
-
-    using FrameworkV1::Microsoft.VisualStudio.TestTools.UnitTesting;
-    using Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices;
-
-    /// <summary>
-    /// The universal test source host tests.
-    /// </summary>
-    [TestClass]
-    public class UniversalTestSourceHostTests
```
Comment (AbhitejJohn) [inline test/UnitTests/PlatformServices.Universal.Unit.Tests/Services/UniversalTestSourceHostTests.cs:15]: >UniversalTestSourceHostTests [](start = 17, length = 28)

why was this removed?

**Comment by AbhitejJohn** [comment]
Do check if we need changes in test.cmd as well and check E2E. This is a great value add!! :)

**Comment by codito** [comment]
What's the rationale for naming all files `ns10` and `ns13`? Isn't a folder serving the purpose?

**Comment by codito** [comment]
@mayankbansal018 can you take a look please. These changes are similar to the PlatformAbstractions layer in TP.

**Comment by AbhitejJohn** [comment]
@codito : It would be way easier to identify them which file we are editing in VS or any other editor.

File: scripts/Build.ps1 (hunk 1 above)
Comment (jayaranigarg) [reply scripts/Build.ps1]: Done

File: scripts/build/TestFx.Settings.targets (hunk 2 above)
Comment (jayaranigarg) [reply scripts/build/TestFx.Settings.targets]: No..removing

**Comment by AbhitejJohn** [comment]
@jayaranigarg : Do update the nuspec files as well - removing dotnet folder, adding this new platform service and adding the new Tracer library as a dependency.

File: src/Adapter/PlatformServices.Shared/netstandard1.0/Services/ns10TraceLogger.cs (hunk 4 above)
Comment (jayaranigarg) [reply src/Adapter/PlatformServices.Shared/netstandard1.0/Services/ns10TraceLogger.cs]: No..removing

File: test/UnitTests/PlatformServices.Shared.Unit.Tests/netstandard1.0/ns10FileOperationsTests.cs (hunk 5 above)
Comment (jayaranigarg) [reply test/UnitTests/PlatformServices.Shared.Unit.Tests/netstandard1.0/ns10FileOperationsTests.cs]: Done.

File: test/UnitTests/PlatformServices.Shared.Unit.Tests/netstandard1.0/ns10FileOperationsTests.cs (hunk 6 above)
Comment (jayaranigarg) [reply test/UnitTests/PlatformServices.Shared.Unit.Tests/netstandard1.0/ns10FileOperationsTests.cs]: Done.

File: test/UnitTests/PlatformServices.Universal.Unit.Tests/Services/UniversalTestSourceHostTests.cs (hunk 9 above)
Comment (jayaranigarg) [reply test/UnitTests/PlatformServices.Universal.Unit.Tests/Services/UniversalTestSourceHostTests.cs:15]: This moved to shared code.

File: src/Adapter/PlatformServices.Shared/netstandard1.0/ns10Constants.cs (hunk 3 above)
Comment (jayaranigarg) [reply src/Adapter/PlatformServices.Shared/netstandard1.0/ns10Constants.cs]: Changed name to Constants.cs

File: test/UnitTests/PlatformServices.NetCore.Unit.Tests/PlatformServices.NetCore.Unit.Tests.csproj [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+  </PropertyGroup>
+  <Import Project="$(TestFxRoot)scripts\build\TestFx.Settings.targets" />
+  <PropertyGroup>
+    <RootNamespace>MSTestAdapter.PlatformServices.NetCore.UnitTests</RootNamespace>
+    <AssemblyName>MSTestAdapter.PlatformServices.NetCore.UnitTests</AssemblyName>
+    <TargetFramework>netcoreapp1.0</TargetFramework>
+    <PackageTargetFallback Condition=" '$(TargetFramework)' == 'netcoreapp1.0' ">$(PackageTargetFallback);portable-net45+win8+wpa81+wp8</PackageTargetFallback>
+  </PropertyGroup>
+  <ItemGroup>
+    <Compile Include="..\MSTest.CoreAdapter.TestUtilities\ActionUtility.cs" Link="ActionUtility.cs" />
```
Comment (smadala) [inline test/UnitTests/PlatformServices.NetCore.Unit.Tests/PlatformServices.NetCore.Unit.Tests.csproj:15]: No need to add cs files in new cs projects.

File: scripts/test.ps1 [hunk 11]
```diff
@@ -49,6 +49,8 @@ $TFT_Configuration = $Configuration
 $TFT_Pattern = $Pattern
 $TFT_Parallel = $Parallel
 $TFT_All = $All
+$TFB_NetCoreContainers =@("MSTestAdapter.PlatformServices.NetCore.UnitTests.dll")
```
Comment (AbhitejJohn) [inline scripts/test.ps1]: nit: would fit better into the env variables on top where all the other patterns are.

File: scripts/test.ps1 [hunk 12]
```diff
@@ -121,23 +138,32 @@ function Invoke-Test
...
-{
+function Run-Test([string[]] $testContainers, [string[]] $netCoreTestContainers)
+{	
     $vstestPath = Get-VSTestPath
-	
-	if(!(Test-Path $vstestPath))
-	{
-		Write-Error "Unable to find vstest.console.exe at $vstestPath. Test aborted."
-	}
+	$TPV2VSTestPath = Get-TPv2VSTestPath
```
Comment (AbhitejJohn) [inline scripts/test.ps1]: nit: local variables start with non-caps.

File: scripts/test.ps1 [hunk 13]
```diff
@@ -146,6 +172,13 @@ function Get-VSTestPath
 	$vstestPath = Join-Path -path $vsInstallPath "Common7\IDE\CommonExtensions\Microsoft\TestWindow\vstest.console.exe"
 	return Resolve-Path -path $vstestPath
 }
+
+function Get-TPv2VSTestPath
+{
+    $packagesPath = Locate-PackagesPath
+	$vstestConsolePath = Join-Path -path $packagesPath "Microsoft.TestPlatform.15.0.1\tools\net46\vstest.console.exe"
```
Comment (AbhitejJohn) [inline scripts/test.ps1]: variable on top please.

Comment (AbhitejJohn) [reply scripts/test.ps1]: This would not work in CDP scenarios where packages do not get copied over.

File: test/UnitTests/PlatformServices.NetCore.Unit.Tests/PlatformServices.NetCore.Unit.Tests.csproj [hunk 14]
```diff
@@ -60,4 +58,9 @@
       <ResolvedCompileFileDefinitions Remove="@(ResolvedCompileFileDefinitions)" Condition="'%(ResolvedCompileFileDefinitions.Filename)' == 'Microsoft.VisualStudio.TestPlatform.TestFramework.Extensions' or '%(ResolvedCompileFileDefinitions.Filename)' == 'Microsoft.VisualStudio.TestPlatform.TestFramework'" />
     </ItemGroup>
   </Target>
+  <Target Name="PostBuild" AfterTargets="PostBuildEvent">
+      <Exec Command="xcopy /Y /I $(TargetDir)..\..\MSTest.CoreAdapter\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.dll $(TargetDir)" />
```
Comment (AbhitejJohn) [inline test/UnitTests/PlatformServices.NetCore.Unit.Tests/PlatformServices.NetCore.Unit.Tests.csproj]: $(OutDir)?

File: scripts/test.ps1 [hunk 15]
```diff
@@ -148,22 +148,28 @@ function Run-Test([string[]] $testContainers, [string[]] $netCoreTestContainers)
...
-	if(!(Test-Path $TPV2VSTestPath))
+	if($testContainers.Count -gt 0)
 	{
-		Write-Error "Unable to find vstest.console.exe at $TPV2VSTestPath. Test aborted."
+		if(!(Test-Path $vstestPath))
+		{
+			Write-Error "Unable to find vstest.console.exe at $vstestPath. Test aborted."
+		}
+	
+		Write-Verbose "$vstestPath $testContainers $additionalArguments /logger:trx"
```
Comment (AbhitejJohn) [inline scripts/test.ps1:159]: TPV2 vstest.console does not work? We might just want to use that everywhere for consistency. You can take this in the next PR.

**Comment by AbhitejJohn** [comment]
Tagging #18 and #19.
